PsfLauncher/ErrorInformation.h

```diff
@@ -0,0 +1,99 @@
+#pragma once
+#include <string>
+#include <windows.h>
+#include <iostream>
+#include "utilities.h"
+
+class ErrorInformation
+{
+public:
+    ErrorInformation()
+    {
+    }
+
+
+    ErrorInformation(std::wstring customMessage, DWORD errorNumber) :
+        ErrorInformation(customMessage, errorNumber, L"")
+    {
+    }
+
+    ErrorInformation(std::wstring customMessage, DWORD errorNumber, std::wstring exeName)
+        : customMessage(customMessage)
+        , errorNumber(errorNumber)
+        , errorMessage(widen(std::system_category().message(errorNumber)))
+        , isThereAnError(true)
+        , exeName(exeName)
+    {
+    }
+
+    ErrorInformation(const ErrorInformation &toCopy)

```

Comment:
Both the copy constructor and copy assignment operator are unnecessary as the compiler will generate them for you. You can remove them

Reply:
When do I need to make a copy and assignment operator then?  I thought I had to make it for all custom objects I made.

Reply:
You need to define a custom copy constructor/assignment operator when your members' copy constructors are not good enough. E.g. if you need to do any resource management.